src/main/java/org/folio/rest/impl/HoldingsRecordsSourceAPI.java

```diff
@@ -0,0 +1,193 @@
+                HoldingsRecordsSources sources = new HoldingsRecordsSources();
+                List<HoldingsRecordsSource> source = reply.result().getResults();
+                sources.setHoldingsRecordsSources(source);
+                sources.setTotalRecords(reply.result().getResultInfo().getTotalRecords());
+                asyncResultHandler.handle(succeededFuture(GetHoldingsRecordsSourcesResponse
+                  .respond200WithApplicationJson(sources)));
+              } else {
+                log.error(reply.cause().getMessage(), reply.cause());
+                asyncResultHandler.handle(succeededFuture(GetHoldingsRecordsSourcesResponse
+                  .respond400WithTextPlain(reply.cause().getMessage())));
+              }
+            });
+        } catch (Exception e) {
+          log.error(e.getMessage(), e);
+          String message = messages.getMessage(lang, MessageConsts.InternalServerError);
+          if (e.getCause() instanceof CQLParseException) {
+            message = " CQL parse error " + e.getLocalizedMessage();
+            asyncResultHandler.handle(succeededFuture(GetHoldingsRecordsSourcesResponse
+              .respond500WithTextPlain(message)));
+          }
+        }
+      });
+  }
+
+  @Override
+  public void postHoldingsRecordsSources(String lang, HoldingsRecordsSource entity,
+    Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+      vertxContext.runOnContext(v -> {
+        try {
+          String id = entity.getId();
+          if (id == null) {
+            id = UUID.randomUUID().toString();
+            entity.setId(id);
+          }
+          String tenantId = TenantTool.tenantId(okapiHeaders);
+          PostgresClient.getInstance(vertxContext.owner(), tenantId).save(REFERENCE_TABLE, id, entity,
+            reply -> {
+              if (reply.succeeded()) {
+                String ret = reply.result();
+                entity.setId(ret);
+                asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(PostHoldingsRecordsSourcesResponse
+                  .respond201WithApplicationJson(entity, PostHoldingsRecordsSourcesResponse
+                  .headersFor201().withLocation(LOCATION_PREFIX + ret))));
+              } else {
+                String msg = PgExceptionUtil.badRequestMessage(reply.cause());
+                msg = (msg == null) ? "Internal server problem: Error message missing" : msg;
+                log.info(msg);
+                asyncResultHandler.handle(Future.succeededFuture(PostHoldingsRecordsSourcesResponse
+                  .respond400WithTextPlain(msg)));
+              }
+            });
+        } catch (Exception e) {
+          log.error(e.getMessage(), e);
+          asyncResultHandler.handle(Future.succeededFuture(PostHoldingsRecordsSourcesResponse
+            .respond500WithTextPlain(messages.getMessage(lang, MessageConsts.InternalServerError))));
+        }
+      });
```

Comment:
Is there any reason why standard `org.folio.rest.persist.PgUtil#post(...)` method can not be used?